compiler/modules/front.syntax/parseStmt.bal

```diff
@@ -116,15 +113,31 @@ function parseStmt(Tokenizer tok) returns Stmt|err:Syntax {
                 return parseMethodCallStmt(tok);
             }
         }
-        [DECIMAL_NUMBER, _]|[STRING_LITERAL, _]|"true"|"false" => {
-            return parseMethodCallStmt(tok);
+        [DECIMAL_NUMBER, _]
+        | [STRING_LITERAL, _]
+        | "true"
+        | "false"
+        | [HEX_INT_LITERAL, _]
+        | [DECIMAL_FP_NUMBER, _, _] => {
+            var peeked = tok.peek();
+            if peeked == "." || peeked == "[" && !check preparseArrayTypeDescRestoreTok(tok) {
+                return parseMethodCallStmt(tok);
+            }
+            return parseVarDeclStmt(tok, startPos);
         }
     }
     return parseError(tok, "unhandled statement");
 }
 
 type AssignOp "="|CompoundAssignOp;
 
+function preparseArrayTypeDescRestoreTok(Tokenizer tok) returns boolean|err:Syntax {
```

Comment:
```
function savePreparseRestore(Tokenizer tok, PreparseFunc func) {
    TokenizerState state = tok.save();
    boolean isTypeDesc = check func(tok);
    tok.restore(state);
    return isTypeDesc;
}
```

Then use with preparseParenTypeDesc also.

}

Reply:
can we use the same helper function with `preparseParenTypeDesc` as there is a `tok.advance()` call after `tok.save()` when its used?